core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -230,8 +231,11 @@ private synchronized void initContext(ClientContext ctx,
     mMasterClientContext = MasterClientContext.newBuilder(ctx)
         .setMasterInquireClient(masterInquireClient).build();
     mMetricsEnabled = getClusterConf().getBoolean(PropertyKey.USER_METRICS_COLLECTION_ENABLED);
-    if (mMetricsEnabled) {
+    if (mMetricsEnabled && CommonUtils.PROCESS_TYPE.get() != CommonUtils.ProcessType.MASTER) {
+      // TODO(lu) report clients metrics in master process
       try {
+        // clients may be used in the master process before master starts.
+        // Retry getting master addresses in this circumstance may block master starting up
         InetSocketAddress masterAddr = masterInquireClient.getPrimaryRpcAddress();
```

Comment:
I don't understand why this would take longer if the process is the master. Is there some kind of race condition?

Reply:
I think that we shouldn't make the master startup depend upon having Alluxio clients available. If anything, I think this should be a change that we make on the catalog and WebServer side to make sure that a client doesn't need to connect before the RPC server starts?